app/src/androidTest/java/com/github/h3lp3rs/h3lp/AwaitHelpActivityTest.kt

```diff
@@ -0,0 +1,267 @@
+        )
+    }
+
+    @Test
+    fun clickingOnAedButtonWorksAndSendsIntent() {
+        clickingOnButtonWorksAndSendsIntent(
+            AedActivity::class.java,
+            withId(R.id.aed_tuto_button), true
+        )
+    }
+
+    @Test
+    fun cancelButtonWorksAndSendsIntent() {
+        clickingOnButtonWorksAndSendsIntent(
+            MainPageActivity::class.java,
+            withId(R.id.cancel_search_button), false
+        )
+    }
+
+    @Test
+    fun getsNotifiedWhenHelpIsComing() {
+        // Forge the right intent
+        val bundle = Bundle()
+
+        bundle.putInt(EXTRA_EMERGENCY_KEY, helpId)
+        bundle.putBoolean(EXTRA_CALLED_EMERGENCIES, true)
+        bundle.putStringArrayList(EXTRA_NEEDED_MEDICATION, arrayListOf(EPIPEN))
+
+        // Setup the database accordingly
+        val emergencyDb = MockDatabase()
+
+        val emergency = EPIPEN_EMERGENCY_INFO
+
+        emergencyDb.setObject(helpId.toString(), EmergencyInformation::class.java, emergency)
+        setDatabase(EMERGENCIES, emergencyDb)
+
+        // Simulate arrival on await page after calling for help
+        launchAndDo(false) {
+            // Nobody coming
+            onView(withId(R.id.incomingHelpersNumber)).check(matches(withText("")))
+
+            // One person is coming
+            val helper1 = Helper(USER_TEST_ID + 1, 2.0, 2.0)
+            val withHelpers = emergency.copy(helpers = ArrayList(listOf(helper1)))
+
+            emergencyDb.setObject(helpId.toString(), EmergencyInformation::class.java, withHelpers)
+
+            onView(withId(R.id.incomingHelpersNumber)).check(
+                matches(
+                    withText(
+                        globalContext.getString(
+                            R.string.one_person_help
+                        )
+                    )
+                )
+            )
+
+            // The same person is coming again, should NOT add a helper to the list
+            emergencyDb.setObject(helpId.toString(), EmergencyInformation::class.java, withHelpers)
+            onView(withId(R.id.incomingHelpersNumber)).check(
+                matches(
+                    withText(
+                        globalContext.getString(
+                            R.string.one_person_help
+                        )
+                    )
+                )
+            )
+
+            // A second person is coming
+            val helper2 = Helper(USER_TEST_ID + 2, 2.1, 2.1)
+            val withMoreHelpers = emergency.copy(helpers = ArrayList(listOf(helper1, helper2)))
+
+            emergencyDb.setObject(
+                helpId.toString(),
+                EmergencyInformation::class.java,
+                withMoreHelpers
+            )
+            onView(withId(R.id.incomingHelpersNumber)).check(
+                matches(
+                    withText(
+                        String.format(
+                            globalContext.getString(R.string.many_people_help), 2
+                        )
```

Comment:
This is great! However, this test is actually testing for numerous test cases, you should have one test per case

Reply:
I didn't touch this test at all since it was never an issue with Cirrus, but you're right I can refactor it :)